src/mame/f256/f256.cpp

```diff
@@ -0,0 +1,2011 @@
+        0xe3, 0xe4, 0xe4, 0xe5, 0xe6, 0xe6, 0xe7, 0xe7, 0xe8, 0xe9, 0xe9, 0xea, 0xea, 0xeb, 0xec, 0xec,
+        0xed, 0xed, 0xee, 0xef, 0xef, 0xf0, 0xf0, 0xf1, 0xf1, 0xf2, 0xf3, 0xf3, 0xf4, 0xf4, 0xf5, 0xf5,
+        0xf6, 0xf7, 0xf7, 0xf8, 0xf8, 0xf9, 0xf9, 0xfa, 0xfb, 0xfb, 0xfc, 0xfc, 0xfd, 0xfd, 0xfe, 0xff
+    };
+
+    memcpy(m_iopage[0]->pointer(), gamma_1_8, 256);
+    memcpy(m_iopage[0]->pointer() + 0x400, gamma_1_8, 256);
+    memcpy(m_iopage[0]->pointer() + 0x800, gamma_1_8, 256);
+    m_video->set_videoram(m_ram->pointer(), m_iopage[0]->pointer(), m_iopage[1]->pointer(), m_iopage[2]->pointer(), m_iopage[3]->pointer());
+    m_video->start();
+
+    // set the current time on the RTC device
+    time_t now;
+    time(&now);
+    system_time stnow = system_time(now);
+
+    m_rtc->set_current_time(stnow);
+
+    // Initialize the VIA0
+    m_via6522[0]->write(via6522_device::VIA_DDRB, 0xFF);  // DDRB
+    m_via6522[0]->write(via6522_device::VIA_DDRA, 0xFF);  // DDRA
+    m_via6522[0]->write(via6522_device::VIA_PB,   0xFF);  // JOYSTICK 2
+    m_via6522[0]->write(via6522_device::VIA_PA,   0xFF);  // JOYSTICK 1
+    m_via6522[0]->write(via6522_device::VIA_DDRB, 0);     // DDRB
+    m_via6522[0]->write(via6522_device::VIA_DDRA, 0);     // DDRA
+
+    // Initialize the VIA1
+    m_via6522[1]->write(via6522_device::VIA_PB, 0);
+    m_via6522[1]->write(via6522_device::VIA_PA, 0);
+    m_via6522[1]->write(via6522_device::VIA_DDRB, 0);     // DDRB
+    m_via6522[1]->write(via6522_device::VIA_DDRA, 0);     // DDRA
+
+    // Initialize SD Card / SPI clock
+    m_spi_clock = timer_alloc(FUNC(f256_state::spi_clock), this);
+    save_item(NAME(m_spi_clock_state));
+	save_item(NAME(m_spi_clock_sysclk));
+	save_item(NAME(m_spi_clock_cycles));
+	save_item(NAME(m_in_bit));
+	save_item(NAME(m_in_latch));
+	save_item(NAME(m_out_latch));
+    save_item(NAME(m_iec_srq));
+
+    m_timer0 = timer_alloc(FUNC(f256_state::timer0), this);
+    m_timer1 = timer_alloc(FUNC(f256_state::timer1), this);
+}
+
+//-------------------------------------------------
+//  device_reset
+//-------------------------------------------------
+void f256_state::device_reset()
+{
+	driver_device::device_reset();
+    reset_mmu();
+    m_via6522[0]->reset();
+    m_via6522[1]->reset();
+
+    m_opl3->reset();
+    m_sdcard->reset();
+    m_spi_clock->adjust(attotime::never);
+	m_spi_clock_cycles = 0;
+    m_in_bit = 0;
+	m_spi_clock_state = false;
+    spi_sd_enabled = 0;
+    m_mouse->reset();
+
+    m_sid[0]->reset();
+    m_sid[1]->reset();
+    m_sn[0]->reset();
+    m_sn[1]->reset();
+    //m_iec->reset();
+    m_uart->reset();
+
+    m_timer0_load = 0;
+    m_timer0_val = 0;
+    m_timer1_load = 0;
+    m_timer1_val = 0;
+}
+
+//-------------------------------------------------
+//  Interrupts
+//-------------------------------------------------
+void f256_state::sof_interrtupt(int state)
+{
+    if (state) // && ((m_interrupt_masks[1] & 0x01) == 0))
```

Comment:
This condition prevents the interrupting device from deasserting the interrupt to the CPU?